packages/ui-components/src/components/toggle-button-group/toggle-button-group.tsx

```diff
@@ -0,0 +1,47 @@
+import { Component, h, Prop, Host, Event, EventEmitter } from '@stencil/core';
+import { IToggleButton } from '../toggle-button/toggle-button.types';
+import { IToggleButtonGroup, IToggleButtonGroupEvents } from './toggle-button-group.types';
+
+@Component({
+	tag: 'kv-toggle-button-group',
+	styleUrl: 'toggle-button-group.scss',
+	shadow: true
+})
+export class KvToggleButtonGroup implements IToggleButtonGroup, IToggleButtonGroupEvents {
+	/** @inheritdoc */
+	@Prop({ reflect: true }) buttons: IToggleButton[] = [];
+	/** @inheritdoc */
+	@Prop({ reflect: true }) withRadio?: boolean = false;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) disabled?: boolean = false;
+	/** @inheritdoc */
+	@Prop({ reflect: true }) selectedButtons?: Record<string, boolean> = {};
+	/** @inheritdoc */
+	@Prop({ reflect: true }) disabledButtons?: Record<string, boolean> = {};
+	/** @inheritdoc */
+	@Prop({ reflect: true }) radioButtons?: Record<string, boolean> = {};
+	/** @inheritdoc */
+	@Event() checkedChange: EventEmitter<string>;
+
+	render() {
+		return (
+			<Host>
+				{this.buttons.map(button => (
+					<kv-toggle-button
+						exportparts="toggle-button"
+						value={button.value}
+						label={button.label}
+						icon={button.icon}
+						download={button.download}
+						href={button.href}
+						target={button.target}
+						preventDefault={button.preventDefault}
+						disabled={this.disabled || this.disabledButtons[button.value]}
+						checked={this.selectedButtons[button.value]}
+						withRadio={this.withRadio || this.radioButtons[button.value]}
```

Comment:
as a component definition, make sense to have only one toggle button with a radio?
Seems like a strange behavior, with a request from design team?

Reply:
I agree that it does not seem that will have a use case with some buttons with toggle and others without.
On the other hand, this approach offers full flexibility to the developer using the component. What do you think?

Reply:
Ok. I understand the flexibility and we can continue with this approach.
In the future we can think if we are using this and take a better decision.